commit 1c856e44774a7ce8eb6dce6828ef689647f83ca4
Author: Wolfgang Bauer <wbauer@tmo.at>
Date:   Wed Nov 6 16:46:09 2013 +0100

    Revert "Files passed as arguments: Ignore unsupported files"
    
    This reverts commit cd9e50ae4f3ded5a78d0cfb09a67684a9c15d726.
    
    See bug#327224 for details.

diff --git a/dolphin/src/dolphinmainwindow.cpp b/dolphin/src/dolphinmainwindow.cpp
index 9da73f9..b477600 100644
--- a/dolphin/src/dolphinmainwindow.cpp
+++ b/dolphin/src/dolphinmainwindow.cpp
@@ -31,7 +31,6 @@
 #include "panels/information/informationpanel.h"
 #include "settings/dolphinsettingsdialog.h"
 #include "statusbar/dolphinstatusbar.h"
-#include "views/dolphinview.h"
 #include "views/dolphinviewactionhandler.h"
 #include "views/dolphinremoteencoding.h"
 #include "views/draganddrophelper.h"
@@ -244,20 +243,8 @@ void DolphinMainWindow::openDirectories(const QList<KUrl>& dirs)
         return;
     }
 
-    // dirs could contain URLs that actually point to archives or other files.
-    // Replace them by URLs we can open where possible and filter the rest out.
-    QList<KUrl> urlsToOpen;
-    foreach (const KUrl& rawUrl, dirs) {
-        const KFileItem& item = KFileItem(KFileItem::Unknown, KFileItem::Unknown, rawUrl);
-        item.determineMimeType();
-        const KUrl& url = DolphinView::openItemAsFolderUrl(item);
-        if (!url.isEmpty()) {
-            urlsToOpen.append(url);
-        }
-    }
-
-    if (urlsToOpen.count() == 1) {
-        m_activeViewContainer->setUrl(urlsToOpen.first());
+    if (dirs.count() == 1) {
+        m_activeViewContainer->setUrl(dirs.first());
         return;
     }
 
@@ -267,12 +254,12 @@ void DolphinMainWindow::openDirectories(const QList<KUrl>& dirs)
 
     // Open each directory inside a new tab. If the "split view" option has been enabled,
     // always show two directories within one tab.
-    QList<KUrl>::const_iterator it = urlsToOpen.begin();
-    while (it != urlsToOpen.end()) {
+    QList<KUrl>::const_iterator it = dirs.begin();
+    while (it != dirs.end()) {
         openNewTab(*it);
         ++it;
 
-        if (hasSplitView && (it != urlsToOpen.end())) {
+        if (hasSplitView && (it != dirs.end())) {
             const int tabIndex = m_viewTab.count() - 1;
             m_viewTab[tabIndex].secondaryView->setUrl(*it);
             ++it;
